lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2305,6 +2313,12 @@ balanceTransaction ctx genChange (ApiT wid) body = do
     txIncoming = decodeTx tl sealedTxIncoming
     toTxOut (TxChange addr amt assets _) =
         TxOut addr (TokenBundle amt assets)
+    getExternalInps (ApiExternalInput (ApiTxIn (ApiT hashTx) ix) (ApiTxOut _ _ (Quantity amt) _)) =
+        (TxIn hashTx ix, Coin $ fromIntegral amt)
+    externalIns = getExternalInps <$> apiExternalInps
+    areOutputsCovered (Tx _ feeM _ _ outs _ _ _) execUnitsCost =
+        sumCoins (map snd $ L.nubBy (\x y -> fst x == fst y) externalIns) ==
+        sumCoins [fromMaybe (Coin 0) feeM, sumCoins $ txOutCoin <$> outs, execUnitsCost]
```

Comment:
Hmmm. knowing if the transaction is balanced is more complicated. There are many parts to the equation, including withdrawals and deposit / refund. I also don't understand the rationale? Why would it be an error to try to balance an already balanced transaction? It should actually work ok as a no-op, and not error out :thinking:

Reply:
I wonder if coin selection survives already balanced tx, there are NonEmpty's used there

Reply:
I will find out